reposerver/repository/repository.go

```diff
@@ -162,27 +163,24 @@ func (s *Service) runRepoOperation(
 		}
 		defer util.Close(closer)
 		return operation(chartPath, revision)
-	}
-	s.repoLock.Lock(gitClient.Root())
-	defer s.repoLock.Unlock(gitClient.Root())
-
-	if !settings.noCache && getCached(revision) {
```

Comment:
check to see if we need this

Reply:
Yes, this is still needed. The cache could be already populated after the repo lock is acquired, so we need to check it second time. https://en.wikipedia.org/wiki/Double-checked_locking